src/components/ShoppingListItem/ShoppingListItem.js

```diff
@@ -1,5 +1,42 @@
-const ShoppingListItem = ({ item }) => {
-  return <li className="shopping-list__item item">{item.itemName}</li>;
+import { useState, useEffect } from 'react';
+
+import isUnder24hSincePurchased from '../../utils/isUnder24hSincePurchased.js';
+
+const ShoppingListItem = ({ listId, itemId, item, handleCheck }) => {
+  const [recentlyPurchased, setRecentlyPurchased] = useState(false);
+
+  // update whether item is recently purchased
+  useEffect(() => {
+    // make sure properties exist and are not null
+    if (
+      'lastPurchaseDate' in item &&
+      item.lastPurchaseDate &&
+      'seconds' in item.lastPurchaseDate
+    )
```

Comment:
Not a must change. [Optional chaining is fully supported now](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). 
```suggestion
    if (item?.lastPurchaseDate?.seconds)
```

Reply:
🤯